services/diddoc/diddoc_query.go

```diff
@@ -160,6 +167,18 @@ func (dd QueryDIDDocRequestService) AreQueryValuesEmpty(c services.ResolverConte
 	return false
 }
 
+func (dd QueryDIDDocRequestService) IsAmbiguousQuery(c services.ResolverContext) bool {
+	// If there is only 1 query parameter and its in AmbiguousQuery, then we need to return InvalidDidUrlError error
+	if len(dd.Queries) == 1 && len(types.ResourceAmbiguousQueries.IntersectWithUrlValues(dd.Queries)) == 1 {
+		return true
+	}
+	// if one query param resouceMetadata is false and its in AmbiguousQuery, then we need to return InvalidDidUrlError error
+	if dd.Queries.Get(types.ResourceMetadata) == "false" && len(types.ResourceAmbiguousQueries.IntersectWithUrlValues(dd.Queries)) == 1 {
+		return true
+	}
+	return false
```

Comment:
What about multiple ambiguous query parameters being used together? E.g., `resourceCollectionId + resourceVersionTime`? You'll be missing these ones out.

Reply:
This has now been handled in `resource_query_metadata_handler.go` line 30-33